catalog/core/catalog-core-localstorageprovider/src/main/java/org/codice/ddf/catalog/content/impl/FileSystemStorageProvider.java

```diff
@@ -388,88 +401,137 @@ public void rollback(StorageRequest request) throws StorageException {
   }
 
   private ContentItem readContent(URI uri) throws StorageException {
-    Path file = getContentFilePath(uri);
+    Path path = getContentFilePath(uri);
 
-    if (file == null) {
+    if (path == null) {
       throw new StorageException(
           "Unable to find file for content ID: " + uri.getSchemeSpecificPart());
     }
 
-    String filename = file.getFileName().toString();
-    String extension = FilenameUtils.getExtension(filename);
-    URI reference = null;
-    // if the file is an external reference, remove the reference extension
-    // so we can get the real extension
-    if (REF_EXT.equals(extension)) {
-      extension = FilenameUtils.getExtension(FilenameUtils.removeExtension(filename));
-      try {
-        reference = new URI(new String(Files.readAllBytes(file), Charset.forName("UTF-8")));
-
-        if (reference.getScheme() == null) {
-          file = Paths.get(reference.toASCIIString());
-        } else if (reference.getScheme().equalsIgnoreCase("file")) {
-          file = Paths.get(reference);
-        } else {
-          file = null;
-        }
-      } catch (IOException | URISyntaxException e) {
-        throw new StorageException(e);
+    String filename = path.getFileName().toString();
+
+    // resolve external reference if necessary, determine the extension, and retrieve an
+    // InputStream to the content
+    InputStream contentInputStream;
+    String extension;
+
+    try {
+      if (REF_EXT.equals(FilenameUtils.getExtension(path.getFileName().toString()))) {
```

Comment:
✏️ Would clarify and simplify the code in this block if you used `filename` as defined on line 411 above.